src/TokenBuilder.php

```diff
@@ -215,16 +245,27 @@ public function setIssuer($issuer)
     /**
      * Add key value pair to payload array
      *
+     * @param array $payload
+     *
      * @return TokenBuilder
      */
-    public function addPayload($key, $value)
+    public function addPayload(array $payload): TokenBuilder
     {
-        $this->payload = array_merge($this->payload, [$key => $value]);
+        if (isset($payload['key']) && isset($payload['value'])) {
+            $this->payload = array_merge($this->payload, [$payload['key'] => $payload['value']]);
 
-        return $this;
+            return $this;
+        }
+
+        throw new TokenBuilderException('Failed to add payload, format wrong. Array must contain key and value.');
     }
 
-    private function encodeHeader()
+    /**
+     * Encode the header string and return it
+     *
+     * @return string
```

Comment:
Would you use @throws phpdoc tag in your docblock comments where it's possible to get exceptions?

Reply:
Another good spot.